dash/backend/src/modules/falco/service/falco.service.ts

```diff
@@ -78,13 +77,29 @@ export class FalcoService {
         return newResultSet;
     }
 
-    async getFalcoCsv( clusterId: number): Promise<FalcoCsvDto> {
-        const queryResponse = await this.falcoDao.getFalcoLogsForExport(clusterId);
+    async getFalcoCsv( clusterId: number,
+                       limit = 20,
+                       page = 0,
+                       priorities?: string [],
+                       orderBy?: string,
+                       startDate?: string,
+                       endDate?: string,
+                       namespace?: string,
+                       pod?: string,
+                       image?: string,
+                       signature?: string,
+                       ): Promise<FalcoCsvDto> {
+
+        // retrieve filtered falco logs and use the query results to build the csv
+        const queryResp = await this.falcoDao.getFalcoLogs(clusterId, limit, page, priorities, orderBy, startDate, endDate, namespace, pod, image, signature);
         const result = [this.csvService.buildLine(['Date', 'Namespace', 'Pod',
             'Image', 'Priority', 'Message'])];
 
-        for (let i = 0; i < queryResponse.logCount; i++) {
-            const falcoCol = queryResponse.fullList[i];
+        // limit to 1000 or less logs
+        let numOfLogs = 0;
+        queryResp.csvLogList.length <= 1000 ? numOfLogs = queryResp.csvLogList.length : numOfLogs = 1000;
```

Comment:
A more common way you see people do this is like this.. 

const numOfLogs = queryResp.csvLogList.length <= 1000 ? queryResp.csvLogList.length : 1000

Also, enforcing the limit here is not ideal because its enforcing it AFTER the query has already happened. Better to enforce the limit higher up in the code prior to the SQL query occurring, that way if they pass in a limit of 100000000000 then it will never actually run against the postgresql server with that limit.

Reply:
I'll keep the falco dao line 105 and the logic here so we can still get download logs up to 1000. "Limit" came through from the frontend as the page size. If we use limit in both of these places, we'll get only 20, 50 or 100 download logs. Let me know what you think.